### PR TITLE
Make createWorkspaceIndexPatterns to handle no default cluster and clean up

### PR DESCRIPTION
**Pull request by ananzh**
### Description
Make createWorkspaceIndexPatterns to handle no default cluster and clean up

### Issues Resolved
NA

## Screenshot
NA

## Changelog

- test: Make createWorkspaceIndexPatterns to handle no default cluster and clean up

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js [hunk 1]
```diff
@@ -3,11 +3,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
-import { WORKSPACE_NAME, DATASOURCE_NAME } from './constants';
+import { WORKSPACE_NAME, DATASOURCE_NAME } from '../../../../../utils/apps/constants';
```
Comment (angle943) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js:6]: should this be `../utils/apps/query_enhancements/constants`?

File: cypress/utils/dashboards/commands.js [hunk 2]
```diff
@@ -45,123 +45,3 @@ Cypress.Commands.add(
...
-    });
-
-    cy.contains('a', indexPattern).click();
-    cy.getElementByTestId('deleteIndexPatternButton').should('be.visible').click();
-    cy.getElementByTestId('confirmModalConfirmButton').should('be.visible').click();
-
-    // wait until delete is done
-    cy.getElementByTestId('headerApplicationTitle').should('be.visible');
-  }
-);
```
Comment (angle943) [inline cypress/utils/dashboards/commands.js:167]: the associated `index.d.ts` can be deleted

Comment (ananzh) [reply cypress/utils/dashboards/commands.js:167]: good call. will update.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js (hunk 1 above)
Comment (ananzh) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js:6]: I put `export * from './query_enhancements/constants';` in the apps constants. Should be good.

**Review by angle943 — APPROVED** [review]

**Review by sejli — DISMISSED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js [hunk 3]
```diff
@@ -3,11 +3,11 @@
...
 import { SECONDARY_ENGINE } from '../../../../../utils/constants';
 
-const miscUtils = new MiscUtils(cy);
+const randomString = Math.random().toString(36).substring(7);
+const workspace = `${WORKSPACE_NAME}-${randomString}`;
```
Comment (wanglam) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js:10]: NIt: Could you elaborate more why we need a random workspace name here? Seems we still using a fixed workspace name for other cases.

**Review by wanglam — APPROVED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js (hunk 3 above)
Comment (angle943) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js:10]: We have our CI/CD tests running in parallel, and if there are tests running concurrently and using the same backend, this can lead to conflicts (for example, one instance deletes a workspace, while another was still running tests for that workspace). We decided that randomizing the workspace name will prevent these edge cases from happening, and we plan on randomizing other test suites as well

Comment (ananzh) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/a_check.spec.js:10]: Yeah it is for future CI/CD.

**Review by LDrago27 — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by LDrago27 — APPROVED** [review]
